### ISSUE TITLE
Rewrite Maven: When parsing a SourceFile of type Maven always try to resolve dependencies in 'repo.maven.apache.org/maven2' repo with JUnit Tests

### ISSUE THREAD

**Issue by ferblaca** [openrewrite/rewrite#1801]
Running openRewrite tests and parse maven sourceFiles with `XmlParser`, if there are dependencies maven always tries to resolve them in:
- file:/home/{USER}/.m2/repository/
- https://repo.maven.apache.org/maven2

This may seem correct, but the problem is that my maven configuration in the `setting.xml` I have other repositories configured:

```
     <repositories>
        <repository>
          <id>inditex-artifacts</id>
          <url>https://my.jfrog.io/my.org/maven-public</url>
          <layout>default</layout>
          <releases>
            <enabled>true</enabled>
          </releases>
          <snapshots>
            <enabled>true</enabled>
          </snapshots>
        </repository>
      </repositories>
```

In fact, if in a Test sourceFile I configure a dependency that neither exists in my local repo nor exists in `https://repo.maven.apache.org/maven2`, since the artifact is deployed in `https://my.jfrog.io/my.org/maven-public`, the error when performing `MavenParser.builder().build().parse(sourceFileMaven)` is as follows:

```
org.openrewrite.maven.internal.MavenDownloadingException: Unable to download dependency from the following repositories :
  - file:/home/fbc/.m2/repository/
  - https://repo.maven.apache.org/maven2
	at org.openrewrite.maven.internal.MavenPomDownloader.download(MavenPomDownloader.java:309)
```

I only reproduce this behaviour when running Tests. However, in a final application using the same maven configuration it is able to resolve the artifacts in my personal repository. Could it be related to the issue #1688 ?

On the other hand, **is there any way that when parsing a Maven SourceFile with dependencies I can exclude the verification that exists and is downloadable from a repository?** The problem is that we want to develop recipes for dependency releases that will be deployed together with the recipes, so those maven dependencies are not yet deployed in our repository, so the Maven parse doesn't give problems sometimes.

Thank you very much!!

**Comment by nmck257** [comment]
Agreed that this looks to overlap w/ #1688 (from the outside at least), and I'd love to see them both resolved (ie consistently use the config from settings.xml wherever possible, vs hardcoded repo locations)

**Comment by tkvangorder** [comment]
Yeah, both of these issues seem to resolve around needing to read the maven settings file. There are slightly different, so I am going to keep both of these issues open. 

@ferblaca The artifacts need to be downloaded from a repository so that we can build an accurate semantic model for the maven project's structure.

**Comment by ferblaca** [comment]
Task **blocked** until OpenRewrite resolves the issue:
- https://github.com/openrewrite/rewrite/issues/1801

This issue prevents the OpenRewrite plugin from reading the maven `settings.xml` configuration file, so:
- If the local repository is configured in a location other than `/home/USER/.m2/repository/`.
- If the remote artifacts are not in `https://repo.maven.apache.org/maven2`
Parsing a maven project will fail the migration recipes as it does not detect the artefacts.

**Comment by traceyyoshima** [comment]
Fixed by #1688

**Comment by rdifrango** [comment]
@traceyyoshima @tkvangorder in what version is this resolved as I'm still seeing this in `4.27.0` and reported it as follows:

https://github.com/openrewrite/rewrite-maven-plugin/issues/400

**Comment by nmck257** [comment]
@ferblaca - let me clarify the fix a bit:

For JUnit tests, you need to opt in to having OpenRewrite locate and parse your `settings.xml`, by setting the property `org.openrewrite.test.readMavenSettingsFromDisk` for your JVM process. We decided against enabling this by default, as the behavior (hunting down things on the filesystem and using it to modify test behavior) could lead to unexpected and unpredictable results in CI/CD environments.

This PR (merged manually) has more context: https://github.com/openrewrite/rewrite/pull/1955

Side note: when running recipes from `rewrite-maven-plugin`, it should "just work", since Maven will already know about your `settings.xml` and will pass that info forward to OpenRewrite.

**Comment by ferblaca** [comment]
> @ferblaca - let me clarify the fix a bit:
> 
> For JUnit tests, you need to opt in to having OpenRewrite locate and parse your `settings.xml`, by setting the property `org.openrewrite.test.readMavenSettingsFromDisk` for your JVM process. We decided against enabling this by default, as the behavior (hunting down things on the filesystem and using it to modify test behavior) could lead to unexpected and unpredictable results in CI/CD environments.
> 
> This PR (merged manually) has more context: #1955
> 
> Side note: when running recipes from `rewrite-maven-plugin`, it should "just work", since Maven will already know about your `settings.xml` and will pass that info forward to OpenRewrite.


Hi @nmck257!

thanks for replying and clarifying the existence of this property, which I have not seen anywhere in the documentation and I think it can be useful for Junit tests (without running the maven plugin).

I've done some testing and it does indeed work by adding the `-Dorg.openrewrite.test.readMavenSettingsFromDisk=true` argument to the maven run. But.... how could I set this property directly in the Test java in the MavenSettings? the API of the RewriteTest.java class was modified and now I don't have it clear...

For example, how could I set this property programmatically in this [test](https://github.com/ferblaca/OpenRewriteRecipe/blob/update_openrewrite_7.29.0/src/test/java/com/ferblaca/openrewrite/recipes/RewriteRecipeMavenTest.java)? 

thank you very much!
